File: XEEPL ERP Backend/src/main/java/com/xeepl/erp/service/SectionService.java
package com.xeepl.erp.service;

import com.xeepl.erp.dto.*;
import com.xeepl.erp.entity.Section;
import com.xeepl.erp.mapper.SectionMapper;
import com.xeepl.erp.repository.SectionRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
@Transactional
public class SectionService {

    private final SectionRepository sectionRepository;

    public SectionDTO createSection(SectionCreateDTO dto) {
        Section s = new Section();
        s.setTitle(dto.getTitle());
        Section saved = sectionRepository.save(s);
        return SectionMapper.toDTO(saved);
    }

    @Transactional(readOnly = true)
    public List<SectionDTO> getAllSections() {
        return sectionRepository.findAll().stream()
                .map(SectionMapper::toDTO)
                .collect(Collectors.toList());
    }

    @Transactional(readOnly = true)
    public SectionDTO getSectionById(Long id) {
        Section s = sectionRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Section not found"));
        return SectionMapper.toDTO(s);
    }

    public SectionDTO updateSection(Long id, SectionUpdateDTO dto) {
        Section s = sectionRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Section not found"));
        if (dto.getTitle() != null) s.setTitle(dto.getTitle());
        Section saved = sectionRepository.save(s);
        return SectionMapper.toDTO(saved);
    }

    public void deleteSection(Long id) {
        sectionRepository.deleteById(id);
    }
}

File: xeepl-erp-frontend/src/services/sectionService.js
import { ENDPOINTS } from '../utils/constants';
import { get, post, put, del } from './api';

export const sectionService = {
  getAllSections: async () => {
    return await get(ENDPOINTS.SECTIONS);
  },

  getSectionById: async (id) => {
    return await get(`${ENDPOINTS.SECTIONS}/${id}`);
  },

  createSection: async (sectionData) => {
    return await post(ENDPOINTS.SECTIONS, sectionData);
  },

  updateSection: async (id, sectionData) => {
    return await put(`${ENDPOINTS.SECTIONS}/${id}`, sectionData);
  },

  deleteSection: async (id) => {
    return await del(`${ENDPOINTS.SECTIONS}/${id}`);
  }
};